### DISCUSSION TITLE
Is it the best solution? What's going on?

### DISCUSSION THREAD

**Discussion by Pleshivtsev** [cssbattle/forum#6] [General]
Vincent De Oliveira Score: 629.09 {275}, 100% match
Is it a cheat?
![solution](https://user-images.githubusercontent.com/33061866/110026832-41315e00-7d42-11eb-8c1d-8eec3df8a742.png)

**Comment by Beowolve** [comment]
The diamond unicode character (⬥) is rendered differently on the server. I don't like that as well. Maybe it would be a good idea to ban unicode characters in the future.  I have seen other target solutions with similar stuff. Vincent De Oliveira seams to be an expert on using them. Just take a look at target #62, his top solution is also using a circle character. There is only a single box rendered with a standard Chrome browser. O_O

**Reply by underdude** [reply]
The difference to my understanding comes from the font that the server uses by default.

Alex reverse-engineered (well, dug the info from few sources) the server setup already a while ago and created a tool to preview what the server actually sees: https://codepen.io/alexzaworski/embed/vwmNKr . Since that is running on a free plan of Google Cloud Functions, it sometimes runs out of quota, and at least few of us have spun our own instances to help finding out these unicode solutions.

**Reply by Pleshivtsev** [reply]
It's not about CSS styling it's more likely about CSS hacking. One of the most important rule - result should be same on different environments. Now we see tuning for one particular environment. It's not good for my point of view.

**Reply by gunnarbittersmann** [reply]
> result should be same on different environments.

That’s not going to work out. Firefox renders thing differently than Chromia; different Chromium versions render things differently. And browsers are not to blame; we’re in quirks modus.

As I said on 2021-02-17 at 07:01 UTC [on Spectrum](https://spectrum.chat/css-battle/general/cssbattle-watercooler~8338bf45-38e9-4aed-bf03-7da2721ea1ab): There might be reasons to not allow characters. Different rendering is none of them.

**Reply by Esger** [reply]
How about serving the server fonts to the client then? - and apply them of course.

**Reply by jbarrantes** [reply]
Agree with @gunnarbittersmann,

Ironically SVG and images are not allowed, this makes sense since this restriction encourages the use of CSS solutions to achieve the desired results.

The case for Unicode is slightly similar but having characters adds to the creativity and hacking that is part of the fun with golfing, still makes competition less appealing for a some people.

Having the option to install the same fonts used by the server would be useful as @Esger proposes, other solutions may be hard to implement or too disruptive.

**Reply by aaronze** [reply]
Based on previous posts from the CSSBattle team, they do want to get rid of unicode support as they feel it doesn't represent the core ideal for css golfing, so I doubt they'll do much to make it more accessible and usable. Their main problem with getting rid of it is how to keep it friendly for golfers. For example just banning the unicode doesn't get the job done, you can still use normal letters like 'O' and 'V'. Banning strings is a challenge and may unintentionally ban other valid solutions like the usage of `:after{content:''}`. It's a complicated problem with no easy fix.

I've done a few unicode solutions myself and while it works it always feels like guessing small changes until it just works and isn't really fun. I agree on removing it, but can't yet see a good way to do so.
